eno-core/src/main/java/fr/insee/eno/core/processing/out/steps/lunatic/table/HeaderCellsProcessing.java

```diff
@@ -0,0 +1,71 @@
+package fr.insee.eno.core.processing.out.steps.lunatic.table;
+
+import fr.insee.eno.core.Constant;
+import fr.insee.eno.core.exceptions.business.UnauthorizedHeaderException;
+import fr.insee.eno.core.model.code.CodeList;
+import fr.insee.eno.core.model.question.EnoTable;
+import fr.insee.eno.core.model.question.TableQuestion;
+import fr.insee.lunatic.model.flat.HeaderType;
+import fr.insee.lunatic.model.flat.LabelType;
+import lombok.Getter;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+public class HeaderCellsProcessing {
+
+    private HeaderCellsProcessing() {}
+
+    public static List<HeaderType> from(EnoTable enoTable, int leftColumnColspan) {
+        flatListCheck(enoTable);
+        return createHeaderCells(enoTable.getHeader(), leftColumnColspan);
+    }
+
+    /**
+     * Complex/nested code lists are not allowed in the header of Lunatic tables.
+     * This method throws an exception if the header code list of the Eno table given is not flat.
+     *  */
+    private static void flatListCheck(EnoTable enoTable) {
+        CodeList headerCodeList = enoTable.getHeader();
+        ComputeCodeListSizes.of(headerCodeList);
+        if (headerCodeList.getMaxDepth() > 0)
+            throw new UnauthorizedHeaderException(enoTable);
+    }
+
+    private static List<HeaderType> createHeaderCells(CodeList headerCodeList, int leftColumnColspan) {
+        List<HeaderType> lunaticHeader = new ArrayList<>();
+        // Top left empty cell
+        addTopLeftCell(leftColumnColspan, lunaticHeader);
+        // Proper header cells
+        addOtherCells(headerCodeList, lunaticHeader);
+        //
+        return lunaticHeader;
+    }
+
+    private static void addTopLeftCell(int leftColumnColspan, List<HeaderType> lunaticHeader) {
+        if (leftColumnColspan == 0)
+            return;
```

Comment:
This "equal to 0" condition is very close to a "magic number". This has a quite sensible logic though -> just add some javadoc

Reply:
https://github.com/InseeFr/Eno/pull/633/commits/2e565c4bc4f5fa574aca1dfa6210b634ca7534cf